pandas/core/indexes/api.py

```diff
@@ -89,13 +110,19 @@ def conv(i):
         index = indexes[0]
         for other in indexes[1:]:
             if not index.equals(other):
+
+                if sort is None:
```

Comment:
I would move ``_unique_indices`` from a nested function to module level (e.g. same as ``_union_indices`` (and maybe conform the spelling indices / indexes), and simply add a ``sort=`` kwarg (which you are already passing into fast_unique_multiple_lists). Then you can do the warning there. just makes this whole function a bit simpler.